src/libply/Makefile.am

```diff
@@ -11,13 +11,17 @@ EXTRA_DIST = grammar.c grammar.h lexer.c lexer.h \
 	arch/aarch64.c \
 	arch/arm.c     \
 	arch/powerpc.c \
-	arch/x86_64.c
+	arch/x86_64.c \
+	arch/mips.c
 
 BUILT_SOURCES = grammar.h lexer.h
 
-libply_la_SOURCES     = 	\
-	arch/@arch@.c		\
-	\
+if MIPS_ARCH
+libply_la_SOURCES     = arch/mips.c
+else
+libply_la_SOURCES     = arch/@arch@.c
+endif
+libply_la_SOURCES    +=	\
```

Comment:
This should not be needed, `@arch@` should already hold the correct value.

Reply:
I remember that a few months ago, configuring with different `--host` parameters, results in different `@arch@`:
with --host=mips-linux-gcc, `@arch@` will be `mips`, while configuring with --host=mipsel-buildroot-linux-musl-gcc, `@arch@` will be `mipsel`. I will try again later, to make sure about it.